tests/test_algorithm.py

```diff
@@ -313,6 +314,47 @@ def handle_data(context, data):
         with self.assertRaises(TypeError):
             algo.run()
 
+    @parameterized.expand([
+        (-1000, 'invalid_base'),
+        (0, 'invalid_base'),
+    ])
+    def test_invalid_capital_base(self, cap_base, name):
+        """
+        Test that the appropriate error is being raised and orders aren't
+        filled for algos with capital base <= 0
+        """
+        algo_text = """
+def initialize(context):
+    pass
+
+def handle_data(context, data):
+    order(sid(24), 1000)
+        """
+
+        sim_params = SimulationParameters(
+            start_session=pd.Timestamp("2006-01-03", tz='UTC'),
+            end_session=pd.Timestamp("2006-01-06", tz='UTC'),
+            capital_base=cap_base,
+            data_frequency="minute",
+            trading_calendar=self.trading_calendar
+        )
+
+        with self.assertRaises(ZeroCapitalError) as exc:
+            # make_algo will trace to TradingAlgorithm,
+            # where the exception will be raised
+            algo = self.make_algo(script=algo_text, sim_params=sim_params)
+        # It shouldn't be able to run the algorithm
+            output = algo.run()
+            # If capital base was less than zero, the none of the orders
+            # should have been filled so starting/ending cash should be
+            # equal
+            expected_cash = [cap_base for x in range(4)]
+            self.assertEqual(expected_cash, list(output['ending_cash']))
+            # Make sure the correct error was raised
+            error = exc.exception
+            self.assertEqual(str(error), 'The initial capital base is set \
```

Comment:
This assertion doesn't match the error being raised in the error class down below. Is this test actually passing?

Reply:
It probably passes because `self.make_algo(...)` raises (as expected), so none of these lines afterwards, but inside the context manager, are executed.

Reply:
Ah, we probably want to outdent this block, and since we save exc as a variable, we can do the checks after.

Reply:
Saw this and took it into account for test_invalid_capital_base. Removed the starting and ending capital base assertion and took the other one out of the context manager